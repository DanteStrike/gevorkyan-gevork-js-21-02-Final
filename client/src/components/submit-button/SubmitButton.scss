.button {
  color: var(--primary-button-font-color);
  background-color: var(--primary-button-background);
  border: none;
  border-radius: 5px;
}
