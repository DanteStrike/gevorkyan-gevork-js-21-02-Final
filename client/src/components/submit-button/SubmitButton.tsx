import React from 'react';
import './SubmitButton.scss';
import {Button as ButtonANDT} from 'antd';

interface ISubmitButtonProps {
  loading?: boolean;
  children?: React.ReactNode;
}

function SubmitButton({children, loading}: ISubmitButtonProps) {
  return (
    <ButtonANDT loading={loading} className="button" type="primary" htmlType="submit" block>
      {children}
    </ButtonANDT>
  );
}

SubmitButton.defaultProps = {
  children: ``,
  loading: false,
};

export default React.memo(SubmitButton);
