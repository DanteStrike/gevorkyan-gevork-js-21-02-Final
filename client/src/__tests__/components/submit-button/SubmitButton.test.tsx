import {shallow} from 'enzyme';
import React from 'react';
import SubmitButton from '../../../components/submit-button/SubmitButton';

describe(`Component SubmitButton should render correctly`, () => {
  it(`snapshot default test`, () => {
    const wrapper = shallow(<SubmitButton />);
    expect(wrapper).toMatchSnapshot();
  });

  it(`snapshot loading test`, () => {
    const wrapper = shallow(<SubmitButton loading />);
    expect(wrapper).toMatchSnapshot();
  });

  it(`snapshot children test`, () => {
    const wrapper = shallow(
      <SubmitButton>
        <div>text</div>
      </SubmitButton>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
