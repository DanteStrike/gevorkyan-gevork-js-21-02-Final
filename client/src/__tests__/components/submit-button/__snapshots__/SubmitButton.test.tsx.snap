// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component SubmitButton should render correctly snapshot children test 1`] = `
<Button
  block={true}
  className="button"
  htmlType="submit"
  loading={false}
  type="primary"
>
  <div>
    text
  </div>
</Button>
`;

exports[`Component SubmitButton should render correctly snapshot default test 1`] = `
<Button
  block={true}
  className="button"
  htmlType="submit"
  loading={false}
  type="primary"
/>
`;

exports[`Component SubmitButton should render correctly snapshot loading test 1`] = `
<Button
  block={true}
  className="button"
  htmlType="submit"
  loading={true}
  type="primary"
/>
`;
